CommonCPPCTest.cmake

```diff
@@ -47,6 +48,13 @@ list(SORT TEST_FILES)
 set(ALL_CPP_TESTS)
 set(ALL_CPP_PERF_TESTS)
 
+set(TEST_ENABLE_BOOST_HEADER "Add -DBOOST_TEST_DYN_LINK to tests" CACHE BOOL ON)
```

Comment:
Fair enough.  So the idea would be to:

#keep legacy behavior
if(NOT DEFINED TEST_ENABLE_BOOST_HEADER)
  set(TEST_ENABLE_BOOST_HEADER ON)
endif()
instead?